src/rulesets/Base/mapreduce.jl

```diff
@@ -75,7 +75,8 @@ function rrule(
 
     function sum_pullback(ȳ)
         call(f, x) = f(x)  # we need to broadcast this to handle dims kwarg
-        f̄_and_x̄s = call.(pullbacks, ȳ)
+        broadcast_ȳ = dims isa Colon  ? Ref(ȳ) : ȳ
+        f̄_and_x̄s = call.(pullbacks, broadcast_ȳ)
```

Comment:
Alternatively, we can do:
```
        f̄_and_x̄s = if dims isa Colon
            map(pb -> pb(ȳ), pullbacks)
        else
            broadcast((pb, ȳi) -> pb(ȳi), pullbacks, ȳ)
        end
```
What think?

Reply:
I'd vote for always using broadcast. And in fact for making it a one-liner, but tastes vary, not important:
```
broadcast((f,z) -> f(z), pullbacks, dims isa Colon ? Ref(dy) : dy)
```